python/lib/core/dmod/core/common/helper_functions.py

```diff
@@ -174,4 +183,323 @@ def merge_dictionaries(first: typing.Mapping = None, second: typing.Mapping = No
+    """
+    Order a given dictionary so that primitive-like values appear in name ordering first, then values without an mro,
+    __slots__, or __dict__ sorted by name, then array-like items sorted by name, then basic object-like items sorted
+    by name, then finally mappings ordered by name.
+
+    Nested dictionaries are also ordered.
+
+    Example:
+        >>> example = {
+            "two": 2.0,
+            "three": False,
+            "one": 1,
+            "four": b"bytes",
+            "five": [1, 2, 3, 4, 5],
+            "six": datetime.now(),
+            "seven": SomeClass(),
+            "eight": {
+                "z": b'z',
+                "a": {
+                    "q": 99,
+                     "f": None
+                },
+                "y": 9
+            }
+        }
+        >>> order_dictionary(example)
+        {
+            "four": b"bytes",
+            "one": 1,
+            "three": False,
+            "two": 2.0,
+            "six": <class datetime>,
+            "seven": <class SomeClass>,
+            "five": [1, 2, 3, 4, 5],
+            "eight": {
+                "y": 9,
+                "z": b'z',
+                "a": {
+                    "f": None,
+                    "q": 99
+                }
+            }
+        }
+
+    Arguments:
+         dictionary: The mapping to order
+
+    Returns:
+        An reording of the dictionary from least complex to theoretically most complex values
+    """
+    primitive_keys: typing.List[str] = list()
+    array_keys: typing.List[str] = list()
+    mapping_keys: typing.List[str] = list()
+    object_keys: typing.List[str] = list()
+    non_mro_keys: typing.List[str] = list()
+
+    for key, value in dictionary.items():
+        if isinstance(key, (str, bytes, bool, numbers.Number)) or value is None:
+            primitive_keys.append(key)
+        elif isinstance(value, typing.Mapping):
+            mapping_keys.append(key)
+        elif isinstance(value, typing.Collection):
+            array_keys.append(key)
+        elif not (hasattr(value, "__mro__") or hasattr(value, "__dict__") or hasattr(value, "__slots__")):
+            non_mro_keys.append(key)
+        else:
+            object_keys.append(key)
+
+    ordered_dictionary = OrderedDict()
+
+    for key in sorted(primitive_keys):
+        ordered_dictionary[key] = dictionary[key]
+
+    for key in sorted(non_mro_keys):
+        ordered_dictionary[key] = dictionary[key]
+
+    for key in sorted(array_keys):
+        ordered_dictionary[key] = dictionary[key]
+
+    for key in sorted(object_keys):
+        ordered_dictionary[key] = dictionary[key]
+
+    for key in sorted(mapping_keys):
+        ordered_dictionary[key] = order_dictionary(dictionary[key])
+
+    return ordered_dictionary
+
+
+def truncate_numbers_in_dictionary(dictionary: typing.Mapping, places: int, copy: bool = False) -> typing.Mapping:
+    if places is None or places <= 0:
+        return dictionary
+
+    if copy:
```

Comment:
It used the comprehension because the input type hint was `typing.Mapping` which doesn't define `copy`, not `dict`. I'm just going to go ahead and change the hinting.